.travis.yml

```diff
@@ -1,54 +1,186 @@
 language: c
+cache: ccache
+os:
+  - linux
+  - osx
 dist: xenial
-
+osx_image: xcode10.2
+compiler:
+  - clang
+  - gcc
 addons:
   apt:
     packages:
-     - cmake
-     - yasm
+      - cmake
+      - yasm
+      - valgrind
+      - libgstreamer-plugins-base1.0-dev
+      - libgstreamer1.0-dev
+      - llvm
+      - clang
   homebrew:
     packages:
+      - ccache
       - yasm
-
-jobs:
+notifications:
+  webhooks:
+    - https://coveralls.io/webhook
+env:
+  - build_type=release
+stages:
+  - name: style
+  - name: test
+  - name: Coveralls And Valgrind
+    if: type != pull_request
+before_install:
+  - "sudo chown -R travis: $HOME/.ccache"
+  - export LD_LIBRARY_PATH=${LD_LIBRARY_PATH}:/usr/local/lib PKG_CONFIG_PATH=${PKG_CONFIG_PATH}:/usr/local/lib/pkgconfig PATH="/usr/local/opt/ccache/libexec:$PATH"
+  - wget -nc https://raw.githubusercontent.com/OpenVisualCloud/SVT-AV1-Resources/master/video.tar.gz || wget -nc http://randomderp.com/video.tar.gz
+  - tar xf video.tar.gz
+  - mkdir -p $TRAVIS_BUILD_DIR/Build/linux/$build_type
+install:
+  - pip install --user cpp-coveralls
+  - export PATH=/Users/travis/Library/Python/2.7/bin:${PATH}
+  #- if [ $TRAVIS_OS_NAME = osx ]; then brew install --HEAD valgrind; fi
+script:
+  - &base_script |
+    cd $TRAVIS_BUILD_DIR/Build/linux/$build_type
+    cmake $TRAVIS_BUILD_DIR -G"Unix Makefiles" -DCMAKE_BUILD_TYPE=$build_type ${CMAKE_EFLAGS[@]}
+    cmake -j $(if [ $TRAVIS_OS_NAME = osx ]; then sysctl -n hw.ncpu; else nproc; fi) --build . &&
+    sudo cmake --build . --target install && cd $TRAVIS_BUILD_DIR
+  - SvtHevcEncApp -encMode 9 -i akiyo_cif.y4m -b test1.h265
+before_cache:
+  - "sudo chown -R travis: $HOME/.ccache"
+matrix:
+  fast_finish: true
+  allow_failures:
+    - name: Style check
+    - name: Binary Identical?
+    - os: osx
+    - name: Valgrind
+    - env: COVERALLS_PARALLEL=true build_type=debug CMAKE_EFLAGS="-DCOVERAGE=ON"
+  exclude:
+    - os: osx
+      compiler: gcc # gcc = clang in macOS, unecessary duplicate
   include:
-   # General Linux build job
-   - name: Build
-     script:
-     - cd Build/linux
-     - ./build.sh release
-   # General macOS build job
-   - name: macOS build
-     os: osx
-     script:
-     - cd Build/linux
-     - ./build.sh release    
-   # Coveralls test job
-   - name: Test & Coveralls
-     before_install:
-     - pip install --user cpp-coveralls
-     script:
-     - cd Build/linux
-     - ./build.sh release
-     after_success:
-     - coveralls
-     
+  # valgrind --tool=memcheck --leak-check=yes --show-reachable=yes ./SvtHevcEncApp -help
+  # Coding style check
+    - name: Style check
+      stage: style
+      addons: skip
+      before_install: skip
+      install: skip
+      script:
+        - git grep -I -n -P "\t|\r| $" -- .  && IS_FAIL=true || true
```

Comment:
What is this line checking? Only "\t\r"? The travis log exceeds the max length.

Reply:
tabs and carriage return and trailing spaces

Reply:
I believe that https://github.com/OpenVisualCloud/SVT-HEVC/pull/226 should fix most of the problems, although maybe not all

Reply:
I have made it a little less verbose, although, the problem files still remain